private/bufpkg/bufmodule/module_file_set.go

```diff
@@ -57,10 +57,27 @@ func newModuleFileSet(
 			),
 		)
 	}
+	multiReadBucket := newMultiModuleReadBucket(moduleReadBuckets...)
+	seenPaths := map[string][]string{}
+	if err := multiReadBucket.WalkModuleFiles(context.Background(), "", func(moduleObjectInfo *moduleObjectInfo) error {
+		modulePrefixedPaths, ok := seenPaths[moduleObjectInfo.Path()]
```

Comment:
In this case, we only ever return an error that reports two duplicate files. This could get frustrating if there are several duplicates: the user has to fix one at a time, re-run the command, and notice others that need to be fixed.

We can restructure this as collecting all of the `seenPaths`, then report _all_ of the paths that collide with a single path. We want the error message to be deterministic, so we'll need to iterate over the map by either walking over all the files again (in a second pass), or by sorting the keys and walking over them in that order.

We might even want to report _all_ instances of `Path` -> `ExternalPath` duplicates. But the main case I can think of if for when users depend on both `buf.build/beta/googleapis` and `buf.build/googleapis/googleapis` - in those cases the number of duplicates would be huge and the user would see a wall of text (which isn't helpful at all).

With that said, I think reporting all duplicates of a single file in a deterministic manner is sufficient. So we'd have errors like (notice that we can also format this as an `[...]`:

```
$ buf build
Failure: pet/v1/pet.proto exists in multiple locations: [paymentapis/pet/v1/pet.proto, buf.build/acme/petapis/petapis/pet/v1/pet.proto, buf.build/acme/something/else/pet/v1/pet.proto].
```

What are your thoughts?

Reply:
Hmm... this brings up an interesting point... and you're right, we probably want to avoid the "`beta/googleapis + googleapis/googleapis` wall of text" situation, so +1 on reporting everything deterministically for duplicates. Let me implement that change.